### DISCUSSION TITLE
Purchase email confirmation never came through - resend also never shows - possible change email address?

### DISCUSSION THREAD

**Discussion by aztec2step** [aclap-dev/video-downloadhelper#994] [Help and Q&A]
### Your question

I just purchased the Premium license and have the activation code. I pasted the code into Video DownloadHelper and it shows "License verified" and displays my correct email address. This means to me that it reached out to your service and validated and also I confirm that I put in my email address correctly. 

I never received the confirmation email which I now understand would include some details of getting support by email. I requested resend of the activation code thinking it would again include this information, but that email never came through either. 

My source issue is that this CoApp seems to work ok on some sites like Rumble.com, but I have not been able to get it to work with YouTube at all. Pretty dissatisfied, but very willing to go through the steps to troubleshoot. Perhaps I can help find the bug. 

I also noticed that it works when I install it on Windows Server 2016, with current version of Firefox, BUT there are no additional addons like "uBlock Origin" and "NoScript". Could these be the issue interfering with the Video DownloadHelper addon? 

Example youtube video that doesn't download (identify downloads) 
https://www.youtube.com/watch?v=Ad8HWFxkCeI
https://www.youtube.com/watch?v=I7mHSEzDs08
are the ones I'm typically interested in, but I also checked a few just going to youtube's home page and they also didn't work. 

### Addon & CoApp version and details

Environment: 
Windows 10 Enterprise - 22H2 - Build 19045.4529
Firefox - 128.0 (64-bit)
Version 9.0.2.12 (stable), target mozilla, locale en-US.
Companion App installed (v2.0.19) Recheck
C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe
License verified

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3880, Firefox 128.0 64-bit, licensed VDH 9.0.2.14a1 beta, CoApp 2.0.19.

I went to your first YouTube video & got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4d05be83-7f83-4fe4-97c6-bc7bee4018d0)

Why do people like to post questions about videos that are hours & hours long?  Rhetorical question.  Ignore it.

As you can see, VDH seems to be recognizing this one just fine.  I dropped down the list of variants & discovered that there is an MKV variant with a higher resolution.  So I clicked that entry.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/87255bb5-fa2a-4c7d-8d03-65111c2fe505)

And now I'm going to download it:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/78121741-5279-4810-9d67-4ac2956efd9a)

I decided to change the file name.  Here it is downloading.

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/87f00626-44fa-42ce-a0ac-993c005838b9)
![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/e3953dcf-8abb-4b6b-93e2-0e71a19add19)

It seems like the larger the file to download, the better service YouTube gives you.  That tells me those wankers are still throttling our downloads.  It's just that @paulrouget Paul has figured out a way to get them to not throttle us as badly as they used to.

I would have started my download of your second example but I assume if I get them both running at the same time, this first one will lose some of its bandwidth.  Besides, this is already running at about 85% of my available bandwidth so there's no percentage in trying to force another download through at the same time.

Oh look.  It's finished.  Crap.  I just noticed I put a `$` in the file name when I meant to use a `#`.  Idiot.  OK.  I renamed the file after the download completed.  Here's what I got.

![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/96a4021e-1e57-4cbb-be44-99e893ac34f6)

And here it is playing in VLC:

![#07](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/62aec3d5-5ef8-43fd-be21-bd46025cced2)

As you can see, it looks just fine.  You'll have to trust me that the audio was equally just fine.  You can also trust me that I did not sit & watch this for over 9 hours.  I just sampled it at approximately 5-minute intervals & there seemed to be video & audio all the way to the end.  I have no idea what this is all about that could possibly be fascinating enough to hold my attention for 9 hours.  I will go to my grave having no idea.

**Reply by GrampaWildWilly** [reply]
And now (with apologies to Monty Python) for something not quite completely different.  Your second video:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/02182608-c74c-4a8a-ab6d-f78a73658ec2)

This one does not have an MKV of a higher resolution.  So I downloaded it.  I won't go through all the same images showing it downloading & all.  But I will show this one.

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/3a621066-8f1d-4de3-a3fb-c9a48473bf3a)

Nowhere near the same download speed because you mercifully chose a reasonable size example this time.  It completed faster than I could capture a Resource Monitor image for it.  Like I said, YouTube is still throttling us.  There's no logical reason this download couldn't have also been much faster.  YouTube is still using some formula to artificially slow down our download speeds.  It's still way better than a couple of years ago, when everything took half its duration no matter what.  But they're still sticking it to us.

The results this time are these:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6b00ed52-f18d-4109-ab2f-f824439d3397)

And . . .

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/626cb650-49f2-4891-9c00-63344dcb7c76)

Once again, I just skimmed it (this time at 1-minute intervals) to make sure I had both video & audio right to the end.  Which I did.

If you look closely at my screenshots of the YouTube pages, you'll note that I do have uBlock Origin enabled & active.  It's not interfering with the correct operation of VDH.

So I don't know why it's not working for you.  It's working for me.  You might want to get the VDH beta to see if that makes a difference for you.

As for your license problem, you'd best open a separate discussion on that subject in the right section of this forum.  Click the Discussions link you can see at the top of this page.  That takes you to the home page of this forum.  There's a couple of boxes at the top of that page.  One of them mentions licenses.  Click that.  Follow the instructions there for dealing with license issues.  Like I said, I don't work for VDH so I can't do anything for you on the subject of your license.

**Comment by jcfrog** [comment]
Hi,
Sorry you did not receive our email, please send us a message at licenses@downloadhelper.net with the email address you used for purchase.
